src/ngraph/op/strided_slice.cpp

```diff
@@ -139,19 +175,40 @@ void op::v1::StridedSlice::validate_and_infer_types()
 
     auto begin_const = as_type_ptr<op::Constant>(input_value(1).get_node_shared_ptr());
     auto end_const = as_type_ptr<op::Constant>(input_value(2).get_node_shared_ptr());
-    auto strides = as_type_ptr<op::Constant>(input_value(3).get_node_shared_ptr());
+    auto strides_const = as_type_ptr<op::Constant>(input_value(3).get_node_shared_ptr());
 
-    if (begin_const && end_const && strides)
+    if (begin_const && end_const && strides_const && data_rank.is_static())
```

Comment:
I checked that IE uses `StridedSliceIE ` to handle `StridedSlice` which uses output shape calculated here.
I've tried to run `Slice` ONNX test on IE backends (using https://github.com/NervanaSystems/ngraph/pull/4419) but I have came across many problems (not related to `StridedSlice`).
@lazarevevgeny Are you able to confirm if it doesn't break anything on IE side?

Reply:
I have created a MR where picked up your branch: https://gitlab-icv.inn.intel.com/inference-engine/dldt/merge_requests/6731

The results of the OpenVINO pre-commit tests will be here: https://openvino-ci.intel.com/job/private-ci/job/gitlab_precommit/4413/

Reply:
Version: nGraph release branch + changes from this branch PASS: https://openvino-ci.intel.com/job/private-ci/job/gitlab_precommit/4540/